packages/iov-bns/src/decode.spec.ts

```diff
@@ -133,11 +149,15 @@ describe("Decode", () => {
         throw new Error("unexpected transaction kind");
       }
       expect(parsed.username).toEqual("bobby");
-      expect(parsed.addresses.size).toEqual(2);
-      expect(parsed.addresses.get("chain1" as ChainId)).toEqual("23456782367823X" as Address);
-      expect(parsed.addresses.get("chain2" as ChainId)).toEqual(
-        "0x001100aabbccddffeeddaa8899776655" as Address,
-      );
+      expect(parsed.addresses.length).toEqual(2);
+      expect(parsed.addresses[0]).toEqual({
+        blockchainId: toUtf8("chain1") as BnsBlockchainId,
```

Comment:
I would almost suggest a function for this...

`blockchainId: chainToBlockchainId("chain1")`